snake:
# Publish all joint states -----------------------------------
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 50
# Position Controllers ---------------------------------------
  joint1_position_controller:
    type: position_controllers/JointPositionController
    joint: joint1
    pid: {p: 10.0, i: 1.0, d: 0.0}

  joint2_position_controller:
    type: position_controllers/JointPositionController
    joint: joint2
    pid: {p: 10.0, i: 1.0, d: 0.0}
    
  joint3_position_controller:
    type: position_controllers/JointPositionController
    joint: joint3
    pid: {p: 10.0, i: 1.0, d: 0.0}

  joint4_position_controller:
    type: position_controllers/JointPositionController
    joint: joint4
    pid: {p: 10.0, i: 1.0, d: 0.0}
    
  joint5_position_controller:
    type: position_controllers/JointPositionController
    joint: joint5
    pid: {p: 10.0, i: 1.0, d: 0.0}

  joint6_position_controller:
    type: position_controllers/JointPositionController
    joint: joint6
    pid: {p: 10.0, i: 1.0, d: 0.0}
    
  joint7_position_controller:
    type: position_controllers/JointPositionController
    joint: joint7
    pid: {p: 10.0, i: 1.0, d: 0.0}

  joint8_position_controller:
    type: position_controllers/JointPositionController
    joint: joint8
    pid: {p: 10.0, i: 1.0, d: 0.0}
    
  joint9_position_controller:
    type: position_controllers/JointPositionController
    joint: joint9
    pid: {p: 10.0, i: 1.0, d: 0.0}

  joint10_position_controller:
    type: position_controllers/JointPositionController
    joint: joint10
    pid: {p: 10.0, i: 1.0, d: 0.0}
    
  joint11_position_controller:
    type: position_controllers/JointPositionController
    joint: joint11
    pid: {p: 10.0, i: 1.0, d: 0.0}

  joint12_position_controller:
    type: position_controllers/JointPositionController
    joint: joint12
    pid: {p: 10.0, i: 1.0, d: 0.0}
    
  joint13_position_controller:
    type: position_controllers/JointPositionController
    joint: joint13
    pid: {p: 10.0, i: 1.0, d: 0.0}

  joint14_position_controller:
    type: position_controllers/JointPositionController
    joint: joint14
    pid: {p: 10.0, i: 1.0, d: 0.0}
      
  joint15_position_controller:
    type: position_controllers/JointPositionController
    joint: joint15
    pid: {p: 10.0, i: 1.0, d: 0.0}

  joint16_position_controller:
    type: position_controllers/JointPositionController
    joint: joint16
    pid: {p: 10.0, i: 1.0, d: 0.0}


  gazebo_ros_control:
    pid_gains:
      joint1:
        p: 10.0
        i: 1.0
        d: 0.0 
      joint2:
        p: 10.0
        i: 1.0
        d: 0.0  
      joint3:
        p: 10.0
        i: 1.0
        d: 0.0 
      joint4:
        p: 10.0
        i: 1.0
        d: 0.0 
      joint5:
        p: 10.0
        i: 1.0
        d: 0.0 
      joint6:
        p: 10.0
        i: 1.0
        d: 0.0  
      joint7:
        p: 10.0
        i: 1.0
        d: 0.0 
      joint8:
        p: 10.0
        i: 1.0
        d: 0.0 
      joint9:
        p: 10.0
        i: 1.0
        d: 0.0 
      joint10:
        p: 10.0
        i: 1.0
        d: 0.0  
      joint11:
        p: 10.0
        i: 1.0
        d: 0.0 
      joint12:
        p: 10.0
        i: 1.0
        d: 0.0 
      joint13:
        p: 10.0
        i: 1.0
        d: 0.0 
      joint14:
        p: 10.0
        i: 1.0
        d: 0.0  
      joint15:
        p: 10.0
        i: 1.0
        d: 0.0 
      joint16:
        p: 10.0
        i: 1.0
        d: 0.0 


    

      
      
    
